lib/mobile_app_backend/global_data_cache.ex

```diff
@@ -0,0 +1,125 @@
+defmodule MobileAppBackend.GlobalDataCache do
+  use GenServer
+  alias MBTAV3API.{JsonApi, Repository}
+  alias MBTAV3API.JsonApi.Object
+
+  @typedoc """
+  A key to disambiguate other persistent_term entries from the one owned by this cache instance.
+  Defaults to the module name.
+  """
+  @type key :: term()
+
+  @type data :: %{
+          lines: Object.line_map(),
+          pattern_ids_by_stop: %{
+            (stop_id :: String.t()) => route_pattern_ids :: [String.t()]
+          },
+          routes: Object.route_map(),
+          route_patterns: Object.route_pattern_map(),
+          stops: Object.stop_map(),
+          trips: Object.trip_map()
+        }
+
+  defmodule State do
+    defstruct [:key, :update_ms]
+  end
+
+  def default_key, do: __MODULE__
+
+  def start_link(opts) do
+    opts = Keyword.merge([key: default_key()], opts)
+    GenServer.start_link(__MODULE__, opts)
+  end
+
+  @spec get_data(key()) :: data()
+  def get_data(key \\ default_key()) do
+    :persistent_term.get(key)
```

Comment:
suggestion: Could this first check if the data is populated and call `update_data` if it hasn't been yet? I think making `update_data` more resilient to failure and ensuring the data is present will make this a more reliable data store that seems safer to call outside the global controller flow.

Reply:
Ensuring the data is present beforehand was why I'm calling `update_data/1` in `init/1`, so that the GenServer doesn't count as started until the data has been loaded, but I guess if that breaks the Docker container then that approach would work too.

I'm not sure what it would mean for `update_data/1` to be more resilient to failure - returning partial data seems like it'd be incorrect, and if it crashes instead, the `GlobalDataCache` will just restart and retry.

Reply:
Yeah agreed partial data wouldn't make sense. If any of the requests fail I would expect nothing to be written to :persistent_term & maybe an error logged. To your point at planning, we aren't doing that in a lot of places right now, but if-minor, would be nice to modify as part of touching this path.

Reply:
Well, the error that's logged will be that the calling process crashed, and we'll get a report in Sentry. In practice, the overwhelming majority of the time, `update_data/1` will be called from the `GlobalDataCache` itself, so there's no way for it to respond with an HTTP error.